cluster/src/main/java/io/atomix/cluster/impl/DefaultClusterService.java

```diff
@@ -154,10 +181,13 @@ private void sendHeartbeat(Endpoint endpoint, byte[] payload) {
   /**
    * Handles a heartbeat message.
    */
-  private void handleHeartbeat(Endpoint endpoint, byte[] message) {
+  private byte[] handleHeartbeat(Endpoint endpoint, byte[] message) {
     ClusterHeartbeat heartbeat = SERIALIZER.decode(message);
     failureDetectors.computeIfAbsent(heartbeat.nodeId(), n -> new PhiAccrualFailureDetector()).report();
     activateNode(new StatefulNode(heartbeat.nodeId(), heartbeat.nodeType(), endpoint));
+    return SERIALIZER.encode(nodes.values().stream()
+        .filter(node -> node.type() == Node.Type.CLIENT)
```

Comment:
Data type?

Reply:
Data nodes are replicated by the `ClusterMetadataService` and would be redundant, so only client nodes are filtered and sent back in response to heartbeats